profiling/src/profile/mod.rs

```diff
@@ -488,6 +489,35 @@ impl Profile {
     pub fn get_string(&self, id: i64) -> Option<&String> {
         self.strings.get_index(id as usize)
     }
+
+    fn get_endpoint_for_label(&self, label: &Label) -> anyhow::Result<&i64> {
+        let local_root_span_id: u64 = if label.str == 0 {
+            /* Safety: the value is a u64, but pprof only has signed values,
+             * so we transmute it; the backend does the same.
+             */
+            unsafe { std::intrinsics::transmute(label.num) }
+        } else {
+            /* Sad path: have to fetch the string from the table and convert
+             * it back into a u64. Clients should change to use numbers!
+             */
+            let string = match self.strings.get_index(label.str as usize) {
+                None => anyhow::bail!(
+                    "failed to retrieve index {} from the string table when looking for endpoint data",
+                    label.str
+                ),
+                Some(string) => string,
+            };
+            string.as_str().parse()?
+        };
```

Comment:
As this PR changes `add_endpoint` to take a u64 and not a String, I'm not quite sure we need this.

E.g., what clients would use the new numberic `add_endpoint` but still send the labels as strings elsewhere?

Reply:
This is coming from a _label_, and there's nothing that stops a client from attaching the label value as a string instead of a number.

Reply:
Hmm... I think having the fallback path is a sharper edge, as it makes things work even when they're not set up the way they should.

E.g., if someone does it wrong, they will still get the right output, which is dangerous...

One suggestion is to consider failing serialization and complaining that this label should not be a string. The `serialize` operation already returns a `Result<EncodedProfile, EncodeError>` so clients already need to handle errors, and perhaps modifying `from` to potentially return an error would allow us to report back what's wrong.

Reply:
Okay, I've  started working on fixing this. There will be two changes:

1. At serialization time, verify that any `local root span id` labels use `.num`.
2. When the sample is added, verify that any `local root span id` labels use `.num`.